pkg/api/types.go

```diff
@@ -878,7 +878,8 @@ type EnvVar struct {
 // EnvVarSource represents a source for the value of an EnvVar.
 // Only one of its fields may be set.
 type EnvVarSource struct {
-	// Selects a field of the pod; only name and namespace are supported.
+	// Selects a field of the pod: supports metadata.name, metadata.namespace, metadata.labels, metadata.annotations,
```

Comment:
Because people reused code that exposed them.

On Fri, Aug 19, 2016 at 2:19 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> Yes we do.
>
> On Fri, Aug 19, 2016 at 2:11 PM, Paul Morie <notifications@github.com>
> wrote:
>
>> In pkg/api/types.go
>> <https://github.com/kubernetes/kubernetes/pull/27880#discussion_r75527150>
>> :
>>
>> > @@ -878,7 +878,8 @@ type EnvVar struct {
>> >  // EnvVarSource represents a source for the value of an EnvVar.
>> >  // Only one of its fields may be set.
>> >  type EnvVarSource struct {
>> > -	// Selects a field of the pod; only name and namespace are supported.
>> > +	// Selects a field of the pod: supports metadata.name, metadata.namespace, metadata.labels, metadata.annotations,
>>
>> @smarterclayton <https://github.com/smarterclayton> we don't currently
>> support labels and annotations via envs :-/
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/27880/files/d3f23004ba0c309298490ce43535eec04fecaf89#r75527150>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABG_p9IJQ7q56l2r7e3Ua-eBnBe2htTCks5qhfG8gaJpZM4I7-Um>
>> .
>>
>
>